src/battle_script_commands.c

```diff
@@ -16861,6 +16861,44 @@ void BS_TryGulpMissile(void)
 }
 
 
+void BS_TryActivateGulpMissile(void)
+{
+    NATIVE_ARGS();
+
+    if (!(gMoveResultFlags & MOVE_RESULT_NO_EFFECT)
+        && !gProtectStructs[gBattlerAttacker].confusionSelfDmg
+        && TARGET_TURN_DAMAGED
+        && gBattleMons[gBattlerTarget].species != SPECIES_CRAMORANT
+        && GetBattlerAbility(gBattlerTarget) == ABILITY_GULP_MISSILE)
+    {
+        if (GetBattlerAbility(gBattlerAttacker) != ABILITY_MAGIC_GUARD)
+        {
+            gBattleMoveDamage = GetNonDynamaxMaxHP(gBattlerAttacker) / 4;
+            if (gBattleMoveDamage == 0)
+                gBattleMoveDamage = 1;
+        }
+        
+        BattleScriptPushCursor();                                                            
+        switch(gBattleMons[gBattlerTarget].species)
+        {
+            case SPECIES_CRAMORANT_GORGING:             
+                TryBattleFormChange(gBattlerTarget, FORM_CHANGE_HIT_BY_MOVE);
+                gBattlescriptCurrInstr = BattleScript_GulpMissileGorging; 
+                break;
+            case SPECIES_CRAMORANT_GULPING: 
+            default:           
```

Comment:
Any particular reason you added a default here?

Reply:
Modern doesn't like switches without defaults

Reply:
Shouldn't the default be a break in this case? Since if it defaults both given cases will be falls and it would be a bug if it starts a form change and goes to the gulp script. 

This could happen in a randomizer when a species other the Cramorant gets Gulp Missile as an ability

Reply:
I think we could use this to have a generic "transformed when hit by move" so that it's not exclusively linked to Gulp Missile.